src/driver-sinowealth.c

```diff
@@ -137,19 +138,39 @@ _Static_assert(sizeof(struct sinowealth_config_report) == SINOWEALTH_CONFIG_SIZE
 struct sinowealth_data {
 	/* this is kinda unnecessary at this time, but all the other drivers do it too ;) */
 	struct sinowealth_config_report config;
+	bool is_sensor_3360;
 };
 
 static int
-sinowealth_raw_to_dpi(int raw)
+sinowealth_raw_to_dpi(struct ratbag_device *device, int raw)
 {
-	return (raw + 1) * 100;
+	struct sinowealth_data *drv_data = device->drv_data;
+	int dpi;
+
+	if (drv_data->is_sensor_3360)
+		dpi = (raw + 1) * 100;
+	else
+		dpi = raw * 100;
```

Comment:
Is this correct? The DM5 is supposed to have a PWM3389, which has a 50 CPI step size.

Reply:
from my testing it seems the 100 CPI step is correct